src/connector.rs

```diff
@@ -85,10 +85,16 @@ where
                 Box::pin(f)
             } else if sch == &http::uri::Scheme::HTTPS {
                 let cfg = self.tls_config.clone();
-                let hostname = match self.override_server_name.as_deref() {
+                let mut hostname = match self.override_server_name.as_deref() {
                     Some(h) => h,
                     None => dst.host().unwrap_or_default(),
                 };
+
+                // Remove square brackets around IPv6 address.
+                if hostname.starts_with('[') && hostname.ends_with(']') && hostname.len() > 2 {
```

Comment:
Any reason we can't use [`trim_matches()`](https://doc.rust-lang.org/std/primitive.str.html#method.trim_matches) here?

Reply:
I wasn't aware of this Rust API. Can update this to 
```
hostname = hostname.trim_matches(|c| c == '[' || c == ']')
```

Reply:
That seems worse as you really only want to trim [ on the left and ] on the right.

Reply:
Good point!
What about
```
hostname = hostname.trim_start_matches('[').trim_end_matches(']')
```
?